src/test/globalSetup.test.ts

```diff
@@ -106,6 +111,12 @@ export const mochaHooks = {
         globals.codelensRootRegistry.dispose()
         globalSandbox.restore()
 
+        // Don't run any validations for tests on telemetry
+        if (!(this.currentTest ?? this.test)?.file?.includes('shared/telemetry/')) {
+            validateTelemetryEmitCalls(this.test as Hook, telemetrySpanSpy)
```

Comment:
can we use assertTelemetry: https://github.com/aws/aws-toolkit-vscode/blob/c6abde34bf9ee5b9a87c6ae883e53ce35a889a2c/src/test/testUtil.ts#L207

that avoids mocks, and checks actual metrics emitted by actual code.

Reply:
assertTelemetry() takes in a metric name, but we want to just in general check if any published metrics meet the requirements.

While we could add a condition to assertTelemetry() that checks if result/reason exist properly, we would miss any current and future test cases where telemetry calls are not explicitly checked for.

Maybe instead I can add a method that can access all metrics published for the session.